verilog/analysis/checkers/dff_name_style_rule_test.cc

```diff
@@ -0,0 +1,310 @@
+      {"module m; always_ff @(posedge c) begin a_ff <= ",
+       {MacroIdentifier, "`DEF"},
+       "; end endmodule"},
+      {"module m; always_ff @(posedge c) begin a_ff <= ",
+       {MacroIdentifier, "`DEF_n"},
+       "; end endmodule"},
+      // Equal base
+      {"module m; always_ff @(posedge c) begin ",
+       {SymbolIdentifier, "a_q"},
+       " <= b_n; end endmodule"},
+  };
+  RunConfiguredLintTestCases<VerilogAnalyzer, DffNameStyleRule>(
+      kTestCases, "output:ff,q;input:n");
+}
+
+TEST(DffNameStyleRuleTest, ExtractPipelineStage) {
+  struct Test {
+    absl::string_view str;
+    std::pair<absl::string_view, std::optional<int>> expected;
+  };
+  const std::vector<Test> kTestCases = {
+      {"data_q0", {"data_q0", {}}}, {"data_q1", {"data_q1", {}}},
+      {"data_q2", {"data_q", {2}}}, {"data_q20", {"data_q", {20}}},
+      {"data_q0", {"data_q0", {}}}, {"a", {"a", {}}},
+      {"data", {"data", {}}}};
+
+  for (auto &test : kTestCases) {
+    auto [result_str, result_int] =
+        DffNameStyleRule::ExtractPipelineStage(test.str);
+    CHECK_EQ(test.expected.first, result_str);
+    CHECK_EQ(test.expected.second == result_int, true);
+  }
+}
+
+TEST(DffNameStyleRuleTest, ExtraChecks) {
+  const std::initializer_list<LintTestCase> kTestCases = {
+      {"wire ", {SymbolIdentifier, "data_ff"}, " = 1;"},
+      {"module m; assign ", {SymbolIdentifier, "data_ff"}, " = 1; endmodule"},
+      {"module m;\nalways_comb begin\n",
+       {SymbolIdentifier, "data_ff"},
+       " = 1;\n",
+       {SymbolIdentifier, "data_ff"},
+       "++;\n",
+       "++",
+       {SymbolIdentifier, "data_ff"},
+       ";\n",
+       {SymbolIdentifier, "data_ff"},
+       " &= 1;\n",
+       "end\nendmodule"},
+      {"reg data_n = 1;"},
+  };
+  RunLintTestCases<VerilogAnalyzer, DffNameStyleRule>(kTestCases);
+}
+
+TEST(DffNameStyleRuleTest, WaiveRegexAccept) {
+  const std::initializer_list<LintTestCase> kTestCases = {
+      {"module m; always_ff @(posedge c) mem <= 1; endmodule"},
+      {"module m; always_ff @(posedge c) mem[addr] <= val; endmodule"},
+      {"module m; always_ff @(posedge c) mem[addr][other] <= val; endmodule"},
+      {"module m; always_ff @(posedge c) if(something) begin mem[addr] <= 0; "
+       "end; endmodule"},
+  };
+  RunLintTestCases<VerilogAnalyzer, DffNameStyleRule>(kTestCases);
+}
+
+TEST(DffNameStyleRuleTest, WaiveIfBranches) {
+  const std::initializer_list<LintTestCase> kTestCases = {
+      {"module m; always_ff @(posedge c) if(!rst_ni) begin a_reg <= 0; end; "
+       "endmodule "},
+      {"module m; always_ff @(posedge c) if( flush_i ) begin a_reg <= 0; end; "
+       "endmodule "},
+      {"module m; always_ff @(posedge c) if(!rst_ni || flush_i) begin a_reg <= "
+       "0; end; "
+       "endmodule "},
+      {"module m; always_ff @(posedge c) if(flush_i || !rst_ni) begin a_reg <= "
+       "0; end; "
+       "endmodule "},
+      {"module m; always_ff @(posedge c) if(reset) begin a_reg <= ",
```

Comment:
I'd also add a test-case with a space between `if` and the open-`(`. It will work as test is post-tokenization, but it is a good documentation.

Reply:
Added 

```
{"module m; always_ff @(posedge c) if ( reset  ) begin a_reg <= ",
 {TK_DecNumber, "0"},
 "; end; endmodule"},
```